### ISSUE TITLE
NAG 7.2 does not build 5.0.2, but does 4.1.6: -pthread passed to ld 

### ISSUE THREAD

**Issue by mathomp4** [open-mpi/ompi#12413]
### What version of Open MPI are you using? (e.g., v4.1.6, v5.0.1, git branch name and hash, etc.)

I'm testing Open MPI 5.0.2 (and 4.1.6)

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

Both installed from tarfile.

Configure command is:

```
$ mkdir build-nag-7.2.01 && cd build-nag-7.2.01
$ ../configure --disable-wrapper-rpath --disable-wrapper-runpath \
      CC=gcc CXX=g++ FC=nagfor \
      --with-pmix=internal --with-hwloc=internal --with-libevent=internal \
      --prefix=/ford1/share/gmao_SIteam/MPI/openmpi/X.Y.Z/nag-7.2.01 |& tee configure.nag-7.2.01.log
```

where `X.Y.Z` is either 4.1.6 or 5.0.2

### Please describe the system on which you are running

* Operating system/version: RHEL8
* Computer hardware: Intel(R) Xeon(R) CPU E5-2680 v4
* Network type: Ethernet (I think? Single node)

-----------------------------

## Details of the problem

I am trying to build Open MPI 5.0.2 with NAG Fortran 7.2.01 (with GCC 12.1 as C/C++) and all works until:
```
make[3]: Entering directory '/ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/ompi/mpi/fortran/mpif-h'
  CCLD     libmpi_mpifh.la
/usr/bin/ld: unrecognized option '-pthread'
/usr/bin/ld: use the --help option for usage information
```
if I build with `V=1`:
```
make[3]: Entering directory '/ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/ompi/mpi/fortran/mpif-h'
/bin/sh ../../../../libtool  --tag=CC   --mode=link gcc  -O3 -DNDEBUG  -finline-functions -mcx16 -version-info 80:0:40  -o libmpi_mpifh.la -rpath /ford1/share/gmao_SIteam/MPI/openmpi/5.0.2/n
ag-7.2.01/lib  ../../../../ompi/libmpi.la ../../../../ompi/mpiext/ftmpi/mpif-h/libmpiext_ftmpi_mpifh.la /ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/opal/libopen-pal.la pr
ofile/libmpi_mpifh_pmpi.la libmpi_mpifh_sizeof.la -lpthread -lrt -lm -lutil  /ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/libevent_core.la
 /ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/libevent_pthreads.la /ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/3rd-party/hwloc-2.7.1/hwloc/libhwloc.la /ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/3rd-party/openpmix/src/libpmix.la
libtool: link: gcc -shared  -fPIC -DPIC  -Wl,--whole-archive ../../../../ompi/mpiext/ftmpi/mpif-h/.libs/libmpiext_ftmpi_mpifh.a profile/.libs/libmpi_mpifh_pmpi.a ./.libs/libmpi_mpifh_sizeof.a -Wl,--no-whole-archive  -Wl,-rpath -Wl,/ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/ompi/.libs -Wl,-rpath -Wl,/ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/opal/.libs -Wl,-rpath -Wl,/ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/3rd-party/openpmix/src/.libs -Wl,-rpath -Wl,/ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/.libs -Wl,-rpath -Wl,/ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/3rd-party/hwloc-2.7.1/hwloc/.libs -Wl,-rpath -Wl,/ford1/share/gmao_SIteam/MPI/openmpi/5.0.2/nag-7.2.01/lib -L/ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/opal/.libs -L/ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/3rd-party/openpmix/src/.libs -L/ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/.libs -L/ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/3rd-party/hwloc-2.7.1/hwloc/.libs ../../../../ompi/.libs/libmpi.so /ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/opal/.libs/libopen-pal.so -lpthread -lrt /ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/3rd-party/openpmix/src/.libs/libpmix.so -lutil /ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/.libs/libevent_core.so /ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/.libs/libevent_pthreads.so /ford1/share/gmao_SIteam/MPI/src/openmpi-5.0.2/build-nag-7.2.01/3rd-party/hwloc-2.7.1/hwloc/.libs/libhwloc.so -lm -ludev -ldl  -O3 -mcx16   -pthread -Wl,-pthread -Wl,-soname -Wl,libmpi_mpifh.so.40 -o .libs/libmpi_mpifh.so.40.40.0
/usr/bin/ld: unrecognized option '-pthread'
/usr/bin/ld: use the --help option for usage information
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:2937: libmpi_mpifh.la] Error 1
```

So, I thought, let me try Open MPI 4.1.6 and:
```
...
make[3]: Entering directory '/ford1/share/gmao_SIteam/MPI/src/openmpi-4.1.6/build-nag-7.2.01/ompi/mpi/fortran/mpif-h'
  GENERATE sizeof_f.f90
  FC       sizeof_f.lo
NAG Fortran Compiler Release 7.2(Shin-Urayasu) Build 7201
[NAG Fortran Compiler normal termination]
  FCLD     libmpi_mpifh_sizeof.la
  CCLD     libmpi_mpifh.la
make[3]: Leaving directory '/ford1/share/gmao_SIteam/MPI/src/openmpi-4.1.6/build-nag-7.2.01/ompi/mpi/fortran/mpif-h'
make[2]: Leaving directory '/ford1/share/gmao_SIteam/MPI/src/openmpi-4.1.6/build-nag-7.2.01/ompi/mpi/fortran/mpif-h'
Making all in mpi/fortran/use-mpi-tkr
make[2]: Entering directory '/ford1/share/gmao_SIteam/MPI/src/openmpi-4.1.6/build-nag-7.2.01/ompi/mpi/fortran/use-mpi-tkr'
  GENERATE mpi-tkr-sizeof.h
  GENERATE mpi-tkr-sizeof.f90
  FC       mpi_aint_add_f90.lo
  FC       mpi_aint_diff_f90.lo
  FC       mpi_comm_spawn_multiple_f90.lo
...
```

It builds successfully.

I looked around here and I did see https://github.com/open-mpi/ompi/issues/4032 which is *old*, but, well, NAG is there, pthread is there... Maybe some old fix for NAG and Open MPI 2 got into 3, stayed around for 4, but got lost with 5?

NOTE: I was going to try and maybe see if I could use our new NAG setup to help with https://github.com/open-mpi/ompi/issues/9795, but that is a configure time fail so...

**Comment by mathomp4** [comment]
Oh, and I can say I've built Open MPI 5.0.2 with NAG 7.2 (and Apple Clang) on my macOS laptop, so I don't think is nagfor per se. Maybe something Clang is happy with but GCC isn't? 🤷🏼

**Comment by bosilca** [comment]
`gcc -dumpspecs | grep pthread` and look for the lines starting with pthread. On my OSX with gcc (12.3.0 installed via macports) I see that `-pthread` will be translated at compile time into `-D_REENTRANT`

**Comment by mathomp4** [comment]
@bosilca Well, on my mac, I build Open MPI with clang as the C compiler, not GCC:
```
❯ clang -dumpspecs | grep pthread
clang: error: unsupported option '-dumpspecs'
clang: error: no input files
```
But, on the RHEL 8 machine I see:
```
@$ gcc -dumpspecs | grep pthread
%{posix:-D_POSIX_SOURCE} %{pthread:-D_REENTRANT}
...
```

Of course, the error seems to say `-pthread` is not a flag `ld` knows (maybe via `-Wl,-pthread` from my verbose make) and:
```
@$ ld -pthread
ld: unrecognized option '-pthread'
ld: use the --help option for usage information
```

It's like configure has decided to pass `-pthread` to `ld` but that's not a flag `ld` knows?

**Comment by ggouaillardet** [comment]
When `nagfor` is used, `-pthread` (that can be pulled from a `.la` file) is replaced with `-Wl,-pthread` via a patch we apply to `config/ltmain.sh`.

In your case, `gcc` is passed both `-pthread` **and** `-Wl,-pthread` which means `ld` will be passed the `-pthread` flag.

I do not quite see the connection between these ...

**Comment by ggouaillardet** [comment]
since you built `4.1.6`, can you please post the output of
```
cd ompi/mpi/fortran/mpif-h
rm libmpi_mpifh.la
make V=1
```

**Comment by ggouaillardet** [comment]
Thanks @mathomp4 

would you mind giving a try with this `4.1.6` tarball?
https://github.com/ggouaillardet/ompi/releases/download/4.1.6/openmpi-4.1.6.tar.bz2

I regenerated a `4.1.6` tarball with the same autotools used by the `v5` branch (they are slightly more recent that the autotools used in the `v4.1` branch.

**Comment by mathomp4** [comment]
@ggouaillardet Well, I'm not sure if it's a good thing or bad thing, but I tried your tarball and it built fine with NAG 7.2.01 on the same system.

**Comment by ggouaillardet** [comment]
I concur, thanks for the test!

**Comment by ggouaillardet** [comment]
I think I am on to something, but I do not quite see a good fix for that...

First the issue can be reproduced by

`configure FC=nagfor && make`
where `nagfor` can be a symbolic link to a working Fortran compiler (e.g. `gfortran)


with `v4.1`, from `ompi/mpi/fortran/mpif-h/profile/Makefile`:
```
LIBS = -lrt -lm -lutil  -lz  -lhwloc  -levent_core -levent_pthreads
```

but from `v5.x`, from `ompi/mpi/fortran/mpif-h/profile/Makefile:

```
LIBS = -lpthread -lrt -lm -lutil -levent_core -levent_pthreads /home/users/u0001043/build/openmpi-5.0.2-nagfor/3rd-party/hwloc-2.7.1/hwloc/libhwloc.la /home/users/u0001043/build/openmpi-5.0.2-nagfor/3rd-party/openpmix/src/libpmix.la
```

the major difference is `v5` explicitly links with `libpmix.la`.
Long story short, `libpmix.la` depends on `libhwloc.la` and `-Wl,-pthread` gets pulled into the generated libraries, that will cause some issue later.

`libpmix.la` nor `libhwloc.la` are not needed here, so not using them might do the trick.


@jsquyres @bwbarrett any ideas on how to best move forward?

**Comment by mathomp4** [comment]
This might be the first ever NAG bug I've had that does require purchasing NAG! 😄 

Thanks for working on this, @ggouaillardet

**Comment by ggouaillardet** [comment]
@mathomp4 can you please give this commit a try?
https://github.com/ggouaillardet/ompi/commit/8d331be905fc033eb53ae3ea655d67df88de95b7

you will need to apply it on a `v5` source tree and they run `autogen.pl` (or `autogen.pl --force` if applying on an official tarball)

Or you might simply download this already bootstrapped tarball at https://github.com/ggouaillardet/ompi/releases/download/nag_pmix/openmpi-gitclone.tar.bz2 and simply run `configure ... && make`

**Comment by mathomp4** [comment]
@ggouaillardet I grabbed the tarball but:

```
...
NAG Fortran Compiler Release 7.2(Shin-Urayasu) Build 7201
[NAG Fortran Compiler normal termination]
  FCLD     libmpi_mpifh_sizeof.la
  CCLD     libmpi_mpifh.la
/usr/bin/ld: unrecognized option '-pthread'
/usr/bin/ld: use the --help option for usage information
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:2938: libmpi_mpifh.la] Error 1
make[3]: Leaving directory '/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/ompi/mpi/fortran/mpif-h'
make[2]: *** [Makefile:3436: all-recursive] Error 1
make[2]: Leaving directory '/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/ompi/mpi/fortran/mpif-h'
make[1]: *** [Makefile:2801: all-recursive] Error 1
make[1]: Leaving directory '/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/ompi'
make: *** [Makefile:1534: all-recursive] Error 1
```

Should I try the source-and-autogen instead?

**Comment by ggouaillardet** [comment]
I am puzzled since I could pass this with my `nagfor` symlink...

could you please `make V=1` and share the output after `CCLD libmpi_mpifh.la`?

**Comment by mathomp4** [comment]
Here is what I see:

```

make[3]: Leaving directory '/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/ompi/mpi/fortran/mpif-h/profile'
make[3]: Entering directory '/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/ompi/mpi/fortran/mpif-h'
/bin/sh ../../../../libtool  --tag=CC   --mode=link gcc  -O3 -DNDEBUG  -finline-functions -mcx16 -version-info 80:0:40  -o libmpi_mpifh.la -rpath /ford1/share/gmao_SIteam/MPI/openmpi/5.0.2-gilles/nag-7.2.01/lib  ../../../../ompi/libmpi.la ../../../../ompi/mpiext/ftmpi/mpif-h/libmpiext_ftmpi_mpifh.la /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/opal/libopen-pal.la profile/libmpi_mpifh_pmpi.la libmpi_mpifh_sizeof.la -lpthread -lrt -lm -lutil  /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/libevent_core.la /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/libevent_pthreads.la /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/hwloc-2.7.1/hwloc/libhwloc.la
libtool: link: gcc -shared  -fPIC -DPIC  -Wl,--whole-archive ../../../../ompi/mpiext/ftmpi/mpif-h/.libs/libmpiext_ftmpi_mpifh.a profile/.libs/libmpi_mpifh_pmpi.a ./.libs/libmpi_mpifh_sizeof.a -Wl,--no-whole-archive  -Wl,-rpath -Wl,/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/ompi/.libs -Wl,-rpath -Wl,/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/opal/.libs -Wl,-rpath -Wl,/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/.libs -Wl,-rpath -Wl,/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/hwloc-2.7.1/hwloc/.libs -Wl,-rpath -Wl,/ford1/share/gmao_SIteam/MPI/openmpi/5.0.2-gilles/nag-7.2.01/lib ../../../../ompi/.libs/libmpi.so /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/opal/.libs/libopen-pal.so -ludev -ldl -lpthread -lrt -lm -lutil /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/.libs/libevent_core.so /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/.libs/libevent_pthreads.so /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/hwloc-2.7.1/hwloc/.libs/libhwloc.so  -O3 -mcx16   -pthread -Wl,-pthread -Wl,-soname -Wl,libmpi_mpifh.so.40 -o .libs/libmpi_mpifh.so.40.40.0
/usr/bin/ld: unrecognized option '-pthread'
/usr/bin/ld: use the --help option for usage information
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:2938: libmpi_mpifh.la] Error 1
make[3]: Leaving directory '/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/ompi/mpi/fortran/mpif-h'
make[2]: *** [Makefile:3436: all-recursive] Error 1
make[2]: Leaving directory '/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/ompi/mpi/fortran/mpif-h'
make[1]: *** [Makefile:2801: all-recursive] Error 1
make[1]: Leaving directory '/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/ompi'
make: *** [Makefile:1534: all-recursive] Error 1
mathomp4@gs6101-bucy .../FromGilles/openmpi-gitclone/build-nag-7.2.01
```

**Comment by ggouaillardet** [comment]
can you please
```
cd ompi/mpi/fortran/mpif-h/profile
rm *.la
make V=1
```
`-Wl,-pthread` does not get pulled with my `nagfor` symlink (!)

**Comment by mathomp4** [comment]
@ggouaillardet If I run those three steps, all is well. 

But if, from my build dir I do:

```
cd ompi/mpi/fortran/mpif-h
rm *.la
make V=1
```
I get the fail:
```
@$ make V=1 |& tee make.log
Making all in profile
make[1]: Entering directory '/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/ompi/mpi/fortran/mpif-h/profile'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/ompi/mpi/fortran/mpif-h/profile'
make[1]: Entering directory '/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/ompi/mpi/fortran/mpif-h'
/bin/sh ../../../../libtool  --tag=FC   --mode=link nagfor  -mismatch   -o libmpi_mpifh_sizeof.la  sizeof_f.lo  -lpthread -lrt -lm -lutil  /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/libevent_core.la /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/libevent_pthreads.la /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/hwloc-2.7.1/hwloc/libhwloc.la
libtool: link: rm -fr  .libs/libmpi_mpifh_sizeof.a .libs/libmpi_mpifh_sizeof.la
libtool: link: ar cr .libs/libmpi_mpifh_sizeof.a .libs/sizeof_f.o
libtool: link: ranlib .libs/libmpi_mpifh_sizeof.a
libtool: link: ( cd ".libs" && rm -f "libmpi_mpifh_sizeof.la" && ln -s "../libmpi_mpifh_sizeof.la" "libmpi_mpifh_sizeof.la" )
/bin/sh ../../../../libtool  --tag=CC   --mode=link gcc  -O3 -DNDEBUG  -finline-functions -mcx16 -version-info 80:0:40  -o libmpi_mpifh.la -rpath /ford1/share/gmao_SIteam/MPI/openmpi/5.0.2-gilles/nag-7.2.01/lib  ../../../../ompi/libmpi.la ../../../../ompi/mpiext/ftmpi/mpif-h/libmpiext_ftmpi_mpifh.la /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/opal/libopen-pal.la profile/libmpi_mpifh_pmpi.la libmpi_mpifh_sizeof.la -lpthread -lrt -lm -lutil  /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/libevent_core.la /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/libevent_pthreads.la /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/hwloc-2.7.1/hwloc/libhwloc.la
libtool: link: gcc -shared  -fPIC -DPIC  -Wl,--whole-archive ../../../../ompi/mpiext/ftmpi/mpif-h/.libs/libmpiext_ftmpi_mpifh.a profile/.libs/libmpi_mpifh_pmpi.a ./.libs/libmpi_mpifh_sizeof.a -Wl,--no-whole-archive  -Wl,-rpath -Wl,/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/ompi/.libs -Wl,-rpath -Wl,/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/opal/.libs -Wl,-rpath -Wl,/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/.libs -Wl,-rpath -Wl,/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/hwloc-2.7.1/hwloc/.libs -Wl,-rpath -Wl,/ford1/share/gmao_SIteam/MPI/openmpi/5.0.2-gilles/nag-7.2.01/lib ../../../../ompi/.libs/libmpi.so /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/opal/.libs/libopen-pal.so -ludev -ldl -lpthread -lrt -lm -lutil /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/.libs/libevent_core.so /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/libevent-2.1.12-stable/.libs/libevent_pthreads.so /ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/3rd-party/hwloc-2.7.1/hwloc/.libs/libhwloc.so  -O3 -mcx16   -pthread -Wl,-pthread -Wl,-soname -Wl,libmpi_mpifh.so.40 -o .libs/libmpi_mpifh.so.40.40.0
/usr/bin/ld: unrecognized option '-pthread'
/usr/bin/ld: use the --help option for usage information
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:2938: libmpi_mpifh.la] Error 1
make[1]: Leaving directory '/ford1/share/gmao_SIteam/MPI/src/FromGilles/openmpi-gitclone/build-nag-7.2.01/ompi/mpi/fortran/mpif-h'
make: *** [Makefile:3436: all-recursive] Error 1
mathomp4@gs6101-bucy .../mpi/fortran/mpif-h
```
So it's like `mpi/fortran/mpif-h/profile` is good but not `mpi/fortran/mpif-h`

**Comment by ggouaillardet** [comment]
Thanks. I expected the three steps to go well, but I wanted to see how  profile/libmpi_mpifh_psizeof.la was built, so can you please share the output?

**Comment by mathomp4** [comment]
Ahhh. My apologies. Here you go:

[profile_make_log.txt](https://github.com/open-mpi/ompi/files/14873048/profile_make_log.txt)

**Comment by ggouaillardet** [comment]
Thanks, unfortunately I could not find anything relevant here ...

Instead of going back and forth, can you upload a tarball of your build directory? If too big, you can simply include all the `config.log`, `Makefile` and `*.la` files. I hope I will find where the `-Wl,-pthread` flags are pulled from.

**Comment by mathomp4** [comment]
@ggouaillardet Here you go. This is my build dir but where I did `--exclude=3rd-party` to the tar command. Figured that was okay for a first try.

[build-nag-7.2.01.tar.gz](https://github.com/open-mpi/ompi/files/14889899/build-nag-7.2.01.tar.gz)

**Comment by ggouaillardet** [comment]
Thanks again!

So I finally found the issue. `-pthread` is pulled from the internal `libevent` library (I failed to see this because I was using my distro's `libevent` that did not provide the `*.la` files.

I uploaded a new tarball (same location) https://github.com/ggouaillardet/ompi/releases/download/nag_pmix/openmpi-gitclone.tar.bz2

could you please give it a try?

**Comment by mathomp4** [comment]
@ggouaillardet I grabbed that tarfile and it configures, it builds, it installs, it checks, and Hello World says "Hello World".[^1]

I think all is good!

[^1]: Well, `use mpi` says "Hello World". [`use mpi_f08` is still broken with NAG](https://github.com/open-mpi/ompi/issues/9795).

**Comment by mathomp4** [comment]
~I believe this has been fixed. I have a *new* NAG bug to report soon.~

Nope. I just tested it and it has not. And I see now that https://github.com/open-mpi/ompi/pull/12458 by @ggouaillardet has not been merged into development yet, so of course it is still an open issue.
